### ISSUE TITLE
Frontend - Update webapp proposal creator to work with Base contracts

### ISSUE THREAD

**Issue by bagelface** [gnars-dao/gnars-frontend#23]
### Description 🗒️

Update webapp [proposal creator](https://www.gnars.wtf/dao/proposals/new) to work with Base contracts.

* * *

### Details 🏗️

* **Update proposal creator:** Update the webapp [proposal creator](https://www.gnars.wtf/dao/proposals/new) to work with the Base contracts. All functionality available on the mainnet proposal creator should be available on the Base proposal creator.
* **Hide the delegation button:** Delegations on Base will require a separate ticket. If that ticket has not been completed, take the opportunity to hide the delegation button so that users are not confused by it. Do not remove it.
* **Buttons require Base connection:** Buttons that send a transaction are updated to not be "active" unless the user is connected to Base network. Otherwise these changes could accidentally send invalid transactions on mainnet and lose gas money.

* * *

### Acceptance Criteria ☑️

1. All of the changes described in the "Details" section above are implemented and tested.
2. Code is clear, clean, and properly documented.
3. UI changes are sensible and non-regressive.

* * *

### Bounty Details 💰
This issue is rewarded by a bounty upon completion. It was estimated by the Gnars Tech Pod as **8 story points** which equates to **10 Gnars or 500 $USDC**.

**Comment by Vayras** [comment]
@bagelface I would like to work on this issue, are there any general guidelines to start working on this? let me know Thanks

**Comment by bagelface** [comment]
> @bagelface I would like to work on this issue, are there any general guidelines to start working on this? let me know Thanks

Hey thanks for the interest. We give users in within the Gnars community first dibs on these issues. We'll leave it open to them for a week and then if no one is interested, it's all yours. If you'd like to join us in the Discord and get priority access read through this [https://github.com/gnars-dao](https://github.com/gnars-dao). We meet every Tuesday at 8pm EST

**Comment by Reubenstein** [comment]
@bagelface Hey I'd be down to work on this. Vlad referred me. I'm millz.eth on discord

**Comment by Reubenstein** [comment]
I got the frontend pulled down and running just fine; seems pretty straightforward, but logically it looks like #18 and #20 should come first. I'd be down to work on #20 but I'll wait to hear back before I start anything. In the meantime I applied to the Tech POD on Discord.

**Comment by bagelface** [comment]
> I got the frontend pulled down and running just fine; seems pretty straightforward, but logically it looks like #18 and #20 should come first. I'd be down to work on #20 but I'll wait to hear back before I start anything. In the meantime I applied to the Tech POD on Discord.

Yea you can definitely take this one. https://github.com/gnars-dao/gnars-frontend/issues/18 definitely should be finished first, but I think this one and https://github.com/gnars-dao/gnars-frontend/issues/20 can be worked on in parallel. Unless, there's something I've overlooked that couples them together somehow? @asmattic had mentioned wanting to grab https://github.com/gnars-dao/gnars-frontend/issues/20 when he finishes up https://github.com/gnars-dao/gnars-frontend/issues/18 on the last Tech Pod call, so if you're cool with grabbing this one that'd work out perfectly.

**Comment by Reubenstein** [comment]
> > I got the frontend pulled down and running just fine; seems pretty straightforward, but logically it looks like #18 and #20 should come first. I'd be down to work on #20 but I'll wait to hear back before I start anything. In the meantime I applied to the Tech POD on Discord.
> 
> Yea you can definitely take this one. #18 definitely should be finished first, but I think this one and #20 can be worked on in parallel. Unless, there's something I've overlooked that couples them together somehow? @asmattic had mentioned wanting to grab #20 when he finishes up #18 on the last Tech Pod call, so if you're cool with grabbing this one that'd work out perfectly.

Great, I'll take this one then. I agree it can mostly be worked on in parallel with #20, the only cross-over I see is the "Buttons require Base connection" part..if there's any common logic between the auction component and the proposal component for the connection I would want to re-use that if possible to keep the code a little bit more DRY.

**Comment by princyym** [comment]
Hello @bagelface, as it has one assigned will my contribution be counts

**Comment by asmattic** [comment]
> > > I got the frontend pulled down and running just fine; seems pretty straightforward, but logically it looks like #18 and #20 should come first. I'd be down to work on #20 but I'll wait to hear back before I start anything. In the meantime I applied to the Tech POD on Discord.
> > 
> > 
> > Yea you can definitely take this one. #18 definitely should be finished first, but I think this one and #20 can be worked on in parallel. Unless, there's something I've overlooked that couples them together somehow? @asmattic had mentioned wanting to grab #20 when he finishes up #18 on the last Tech Pod call, so if you're cool with grabbing this one that'd work out perfectly.
> 
> Great, I'll take this one then. I agree it can mostly be worked on in parallel with #20, the only cross-over I see is the "Buttons require Base connection" part..if there's any common logic between the auction component and the proposal component for the connection I would want to re-use that if possible to keep the code a little bit more DRY.

That sounds good to me. Now that (#18)[https://github.com/gnars-dao/gnars-frontend/issues/18] is wrapped up I'm good to start work on (#20)[https://github.com/gnars-dao/gnars-frontend/issues/20] if there are no objections and @Reubenstein and I can communicate and keep up to date on any potentially shared logic in parallel.

**Comment by Reubenstein** [comment]
> > > > I got the frontend pulled down and running just fine; seems pretty straightforward, but logically it looks like #18 and #20 should come first. I'd be down to work on #20 but I'll wait to hear back before I start anything. In the meantime I applied to the Tech POD on Discord.
> > > 
> > > 
> > > Yea you can definitely take this one. #18 definitely should be finished first, but I think this one and #20 can be worked on in parallel. Unless, there's something I've overlooked that couples them together somehow? @asmattic had mentioned wanting to grab #20 when he finishes up #18 on the last Tech Pod call, so if you're cool with grabbing this one that'd work out perfectly.
> > 
> > 
> > Great, I'll take this one then. I agree it can mostly be worked on in parallel with #20, the only cross-over I see is the "Buttons require Base connection" part..if there's any common logic between the auction component and the proposal component for the connection I would want to re-use that if possible to keep the code a little bit more DRY.
> 
> That sounds good to me. Now that #18 is wrapped up I'm good to start work on #20 if there are no objections and @Reubenstein and I can communicate and keep up to date on any potentially shared logic in parallel.

Sounds good, I'll get started on this one today.

**Comment by Emmanueltech** [comment]
Hi is this task open I would like to work on it?

**Comment by bagelface** [comment]
> Hi is this task open I would like to work on it?

Potentially. Please join us on our Tech Pod call tomorrow at 20:00 UTC-4 (EST) and we can discuss.
